File: DSAlgorithms/Leetcode/c++/sliding_window/219-contains_duplicates_II.cpp
class Solution {
public:
    bool containsNearbyDuplicate(vector<int>& nums, int k) {
       std::unordered_set<int> window;
       int left = 0;
       for(int right = 0; right < nums.size(); right++) 
       {
           if(right - left > k)
            {
                window.erase(nums[left]);
                left++;
            }
            std::unordered_set<int>::iterator set_iterator = window.find(nums[right]);
            if(set_iterator != window.end())
                return true;
            window.insert(nums[right]);
       }
       return false;
    }
};

File: DSAlgorithms/Leetcode/ts/src/sliding_window/219-contains_duplicates_II.ts
const containsNearbyDuplicate = (nums: number[], k: number): boolean =>{
    const window = new Set<number>();
    let left = 0;
    for(let right = 0; right < nums.length; right++)
    {
        if(right - left  > k)
        {
            window.delete(nums[left]);
            left++;
        }
        if(window.has(nums[right]))
            return true;
        window.add(nums[right]);
    }
    return false;

};